lib/project_types/script/layers/infrastructure/task_runner.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      class TaskRunner
+        TASK_RUNNERS = {
+          "ts" => Infrastructure::AssemblyScriptTaskRunner,
+        }
+
+        def self.for(ctx, language)
+          raise Errors::TaskRunnerNotFoundError unless TASK_RUNNERS[language]
+          TASK_RUNNERS[language].new(ctx)
+        end
+      end
+    end
+  end
+end
```

Comment:
I think this `TaskRunner` should be the base class for all such task runners, and hence should be inherited by `AssemblyScriptTaskRunner`.  A `TaskRunner` should have very clear contract on what its main methods are.  I am just referring back to the original discussion [here](https://github.com/Shopify/script-service/issues/1041).  The main methods can be: `initialize`, `build`, `dependencies_installed?`, `install_dependencies`, and etc.  This way, anyone creating a subclass knows exactly what methods they must implement.

Reply:
BTW, are you planning on adding the `initialize` logic in the TaskRunner as per original discussion to also create the `tsconfig.json`?  Also, FYI of Saul's [work](https://shopify.slack.com/archives/C013V2UL6LC/p1592229001000900) to build the AS language plugin, and when it's in, we probably don't need any `tsconfig.json`?  But, none the less, the concept of the `initialize` step to create any config file could be a main method here.

Reply:
Yeah I agree, thats part of my plan! This PR basically just renames `ScriptBuilder` => `TaskRunner` and adds in the appropriate other methods.

My plan is to create another class like AssemblyScriptProject that knows about the TaskRunner and the ProjectCreate, so we can get rid of this pattern where we do `Class.for(something)`. Then improve the interface of these classes so subclasses know what to implement. I was just going to be consistent for now, then open a PR to change all of these instances as it would be cleaner once I have the `AssemblyScriptProject` class.

> BTW, are you planning on adding the initialize logic in the TaskRunner as per original discussion to also create the tsconfig.json

I personally see that logic as being creation logic, which should live in the `ProjectCreator`. But I'm open to discussion if you think otherwise

Reply:
Sounds like a plan.  I think what you have so far is great!  Once we see `ProjectCreator`, we can see if there is any other tweaks we need.  👍